src/main/scala/se/lu/nateko/cp/meta/routes/SitemapRoute.scala

```diff
@@ -2,23 +2,67 @@ package se.lu.nateko.cp.meta.routes
 
 import akka.http.scaladsl.model.*
 import akka.http.scaladsl.server.Directives.*
-import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.{ExceptionHandler, Route}
 import se.lu.nateko.cp.meta.api.SparqlRunner
 import se.lu.nateko.cp.meta.core.data.{EnvriConfig, EnvriConfigs, envriConf}
+import java.net.URI
 import se.lu.nateko.cp.meta.services.metaexport.SchemaOrg
+import se.lu.nateko.cp.meta.core.data.CountryCode
 
 object SitemapRoute {
+
+	val SiteMap = "sitemap.xml"
+	val DataSitemap = "data-sitemap.xml"
+	val CollectionsSitemap = "collections-sitemap.xml"
+	val DocumentsSitemap = "documents-sitemap.xml"
+
+	private val exceptionHandler = ExceptionHandler{
+
+		case argErr: IllegalArgumentException =>
+			complete(StatusCodes.NotFound)
+
+		case err => throw err
+	}
+
 	def apply(sparqler: SparqlRunner)(using EnvriConfigs): Route = {
 		val extractEnvri = AuthenticationRouting.extractEnvriDirective
 
-		(get & path("sitemap.xml") & extractEnvri){ implicit envri =>
+		(get & extractEnvri){ implicit envri =>
 			given EnvriConfig = envriConf
-			complete(
-				HttpEntity(
-					ContentType.WithCharset(MediaTypes.`text/xml`, HttpCharsets.`UTF-8`),
-					views.xml.Sitemap(SchemaOrg.dataObjs(sparqler) ++ SchemaOrg.collObjs(sparqler) ++ SchemaOrg.docObjs(sparqler)).body
+			path(SiteMap):
+				val sitemaps = Seq(
+					URI(s"https://${envriConf.metaHost}/${DataSitemap}"),
+					URI(s"https://${envriConf.metaHost}/${CollectionsSitemap}"),
+					URI(s"https://${envriConf.metaHost}/${DocumentsSitemap}"))
+
+				complete(
+					HttpEntity(
+						ContentType.WithCharset(MediaTypes.`text/xml`, HttpCharsets.`UTF-8`),
+						views.xml.IndexSitemap(sitemaps).body
+					)
 				)
-			)
+			~
+			path("""([a-z]{2})-data-sitemap.xml""".r):
+				case countryCode if CountryCode.unapply(countryCode.toUpperCase).isDefined =>
+					handleExceptions(exceptionHandler){completeRequest(SchemaOrg.dataObjsByCountry(sparqler, countryCode.toUpperCase))}
+				case _ => reject
```

Comment:
I believe we can use pattern matching on CountryCode instead, and not deal with uppercasing here:
```suggestion
			path("""([a-z]{2})-data-sitemap.xml""".r):
				case CountryCode(code) =>
					handleExceptions(exceptionHandler) {
						completeRequest(SchemaOrg.dataObjsByCountry(sparqler, code.code))
					}
				case _ => reject
```